CRITICAL - 2022-01-25 00:06:53 --> Argument 1 passed to App\Libraries\GroceryCrud::callbackInsert() must be callable, string given, called in D:\xampp\htdocs\tokobuku\app\Controllers\User.php on line 32
#0 D:\xampp\htdocs\tokobuku\app\Controllers\User.php(32): App\Libraries\GroceryCrud->callbackInsert('save')
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): App\Controllers\User->index('add')
#2 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\User))
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#4 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#5 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#6 {main}
ERROR - 2022-01-25 00:12:40 --> Table 'tokobuku.user' doesn't exist
CRITICAL - 2022-01-25 00:12:40 --> Table 'tokobuku.user' doesn't exist
#0 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php(292): mysqli->query('SELECT 1\nFROM `...', 0)
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(670): CodeIgniter\Database\MySQLi\Connection->execute('SELECT 1\nFROM `...')
#2 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(598): CodeIgniter\Database\BaseConnection->simpleQuery('SELECT 1\nFROM `...')
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1455): CodeIgniter\Database\BaseConnection->query('SELECT 1\nFROM `...', Array, false)
#4 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Rules.php(142): CodeIgniter\Database\BaseBuilder->get()
#5 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Validation.php(292): CodeIgniter\Validation\Rules->is_unique('admin', 'username', Array, NULL)
#6 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Validation.php(157): CodeIgniter\Validation\Validation->processRules('username', 'Username', 'admin', Array, Array)
#7 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(685): CodeIgniter\Validation\Validation->run(Array)
#8 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(4295): App\Libraries\grocery_CRUD_Model_Driver->db_insert_validation()
#9 D:\xampp\htdocs\tokobuku\app\Controllers\User.php(44): App\Libraries\GroceryCrud->render()
#10 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): App\Controllers\User->index('insert_validati...')
#11 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\User))
#12 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#13 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#14 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#15 {main}
ERROR - 2022-01-25 00:12:46 --> Table 'tokobuku.user' doesn't exist
CRITICAL - 2022-01-25 00:12:46 --> Table 'tokobuku.user' doesn't exist
#0 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php(292): mysqli->query('SELECT 1\nFROM `...', 0)
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(670): CodeIgniter\Database\MySQLi\Connection->execute('SELECT 1\nFROM `...')
#2 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(598): CodeIgniter\Database\BaseConnection->simpleQuery('SELECT 1\nFROM `...')
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1455): CodeIgniter\Database\BaseConnection->query('SELECT 1\nFROM `...', Array, false)
#4 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Rules.php(142): CodeIgniter\Database\BaseBuilder->get()
#5 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Validation.php(292): CodeIgniter\Validation\Rules->is_unique('admin', 'username', Array, NULL)
#6 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Validation.php(157): CodeIgniter\Validation\Validation->processRules('username', 'Username', 'admin', Array, Array)
#7 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(685): CodeIgniter\Validation\Validation->run(Array)
#8 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(4295): App\Libraries\grocery_CRUD_Model_Driver->db_insert_validation()
#9 D:\xampp\htdocs\tokobuku\app\Controllers\User.php(44): App\Libraries\GroceryCrud->render()
#10 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): App\Controllers\User->index('insert_validati...')
#11 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\User))
#12 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#13 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#14 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#15 {main}
CRITICAL - 2022-01-25 00:15:42 --> edit_unique is not a valid rule.
#0 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Validation.php(300): CodeIgniter\Validation\Exceptions\ValidationException::forRuleNotFound('edit_unique')
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Validation\Validation.php(157): CodeIgniter\Validation\Validation->processRules('username', 'Username', 'admin12', Array, Array)
#2 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(794): CodeIgniter\Validation\Validation->run(Array)
#3 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(4302): App\Libraries\grocery_CRUD_Model_Driver->db_update_validation()
#4 D:\xampp\htdocs\tokobuku\app\Controllers\User.php(45): App\Libraries\GroceryCrud->render()
#5 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): App\Controllers\User->index('update_validati...', '4')
#6 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\User))
#7 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#9 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#10 {main}
CRITICAL - 2022-01-25 00:32:07 --> The table name does not exist. Please check you database and try again.
#0 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(4143): App\Libraries\GroceryCrud->get_table()
#1 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(4160): App\Libraries\GroceryCrud->pre_render()
#2 D:\xampp\htdocs\tokobuku\app\Controllers\Employees.php(47): App\Libraries\GroceryCrud->render()
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): App\Controllers\Employees->index()
#4 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Employees))
#5 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#6 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#7 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#8 {main}
ERROR - 2022-01-25 01:20:50 --> Table 'tokobuku.level_id' doesn't exist
CRITICAL - 2022-01-25 01:20:50 --> Table 'tokobuku.level_id' doesn't exist
#0 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php(292): mysqli->query('SHOW COLUMNS FR...', 0)
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(670): CodeIgniter\Database\MySQLi\Connection->execute('SHOW COLUMNS FR...')
#2 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(598): CodeIgniter\Database\BaseConnection->simpleQuery('SHOW COLUMNS FR...')
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php(402): CodeIgniter\Database\BaseConnection->query('SHOW COLUMNS FR...')
#4 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(1436): CodeIgniter\Database\MySQLi\Connection->_fieldData('`level_id`')
#5 D:\xampp\htdocs\tokobuku\app\Models\GroceryCrudModel.php(478): CodeIgniter\Database\BaseConnection->getFieldData('level_id')
#6 D:\xampp\htdocs\tokobuku\app\Models\GroceryCrudModel.php(567): App\Models\GroceryCrudModel->get_field_types('level_id')
#7 D:\xampp\htdocs\tokobuku\app\Models\GroceryCrudModel.php(212): App\Models\GroceryCrudModel->get_primary_key('level_id')
#8 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(1198): App\Models\GroceryCrudModel->join_relation('level_id', 'level_id', 'level_name')
#9 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(1340): App\Libraries\grocery_CRUD_Model_Driver->get_list()
#10 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(4185): App\Libraries\grocery_CRUD_Layout->showList(false, Object(stdClass))
#11 D:\xampp\htdocs\tokobuku\app\Controllers\Employees.php(51): App\Libraries\GroceryCrud->render()
#12 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): App\Controllers\Employees->index()
#13 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Employees))
#14 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#15 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#16 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#17 {main}
ERROR - 2022-01-25 02:24:10 --> Unknown column 'deleted_at' in 'where clause'
CRITICAL - 2022-01-25 02:24:10 --> Unknown column 'deleted_at' in 'where clause'
#0 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php(292): mysqli->query('SELECT `custome...', 0)
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(670): CodeIgniter\Database\MySQLi\Connection->execute('SELECT `custome...')
#2 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseConnection.php(598): CodeIgniter\Database\BaseConnection->simpleQuery('SELECT `custome...')
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1455): CodeIgniter\Database\BaseConnection->query('SELECT `custome...', Array, false)
#4 D:\xampp\htdocs\tokobuku\app\Models\GroceryCrudModel.php(89): CodeIgniter\Database\BaseBuilder->get()
#5 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(1223): App\Models\GroceryCrudModel->get_list()
#6 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(1340): App\Libraries\grocery_CRUD_Model_Driver->get_list()
#7 D:\xampp\htdocs\tokobuku\app\Libraries\GroceryCrud.php(4185): App\Libraries\grocery_CRUD_Layout->showList(false, Object(stdClass))
#8 D:\xampp\htdocs\tokobuku\app\Controllers\Customer.php(43): App\Libraries\GroceryCrud->render()
#9 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): App\Controllers\Customer->index()
#10 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Customer))
#11 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#12 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#13 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#14 {main}
ERROR - 2022-01-25 20:21:50 --> Error connecting to the database: Class 'SQLite3' not found
CRITICAL - 2022-01-25 20:39:51 --> Class name must be a valid object or a string
#0 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Config\BaseService.php(248): Myth\Auth\Config\Services::activator(Object(Myth\Auth\Config\Auth), false)
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Config\BaseService.php(189): CodeIgniter\Config\BaseService::__callStatic('activator', Array)
#2 D:\xampp\htdocs\tokobuku\vendor\myth\auth\src\Config\Services.php(86): CodeIgniter\Config\BaseService::getSharedInstance('activator', NULL)
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Config\BaseService.php(248): Myth\Auth\Config\Services::activator()
#4 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Common.php(929): CodeIgniter\Config\BaseService::__callStatic('activator', Array)
#5 D:\xampp\htdocs\tokobuku\vendor\myth\auth\src\Controllers\AuthController.php(195): service('activator')
#6 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): Myth\Auth\Controllers\AuthController->attemptRegister()
#7 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(Myth\Auth\Controllers\AuthController))
#8 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#10 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#11 {main}
CRITICAL - 2022-01-25 20:43:39 --> Class name must be a valid object or a string
#0 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Config\BaseService.php(248): Myth\Auth\Config\Services::activator(Object(Myth\Auth\Config\Auth), false)
#1 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Config\BaseService.php(189): CodeIgniter\Config\BaseService::__callStatic('activator', Array)
#2 D:\xampp\htdocs\tokobuku\vendor\myth\auth\src\Config\Services.php(86): CodeIgniter\Config\BaseService::getSharedInstance('activator', NULL)
#3 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Config\BaseService.php(248): Myth\Auth\Config\Services::activator()
#4 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Common.php(929): CodeIgniter\Config\BaseService::__callStatic('activator', Array)
#5 D:\xampp\htdocs\tokobuku\vendor\myth\auth\src\Controllers\AuthController.php(195): service('activator')
#6 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(825): Myth\Auth\Controllers\AuthController->attemptRegister()
#7 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(412): CodeIgniter\CodeIgniter->runController(Object(Myth\Auth\Controllers\AuthController))
#8 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\CodeIgniter.php(320): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 D:\xampp\htdocs\tokobuku\public\index.php(37): CodeIgniter\CodeIgniter->run()
#10 D:\xampp\htdocs\tokobuku\vendor\codeigniter4\framework\system\Commands\Server\rewrite.php(43): require_once('D:\\xampp\\htdocs...')
#11 {main}
ERROR - 2022-01-25 22:38:11 --> Error connecting to the database: Class 'SQLite3' not found
